client/blb/client.go

```diff
@@ -1118,6 +1118,104 @@ type tractResultRepl struct {
 	thisB []byte
 }
 
+func (cli *Client) readOneTractWithResult(
+	ctx context.Context,
+	host string,
+	reqID string,
+	tract *core.TractInfo,
+	thisB []byte,
+	thisOffset int64) tractResultRepl {
+	var badVersionHost string
+	// TODO(eric): fill in otherHosts when ts-ts cancellation is done.
+	thisBLoc, err := cli.tractservers.Read(ctx, host, nil, reqID, tract.Tract, tract.Version, len(thisB), thisOffset)
+	if err == core.ErrVersionMismatch {
+		// TODO(eric): fix this check when we redo error reporting.
+		badVersionHost = host
+	}
+	readDone()
+	if err != core.NoError && err != core.ErrEOF {
+		// TODO(eric): redo bad TS reporting mechanism.
+		return tractResultRepl{
+			tractResult: tractResult{0, 0, err, badVersionHost},
+		}
+	}
+	log.V(1).Infof("read %s from tractserver at address %s: %s", tract.Tract, host, err)
+
+	return tractResultRepl{
+		tractResult: tractResult{len(thisB), len(thisBLoc), err, badVersionHost},
+		thisB:       thisBLoc,
+	}
+}
+
+// vars for test injection
+var (
+	randOrder         = getRandomOrder
+	backupRequestFunc = doParallelBackupReads
+	readDone          = func() {} // call when read/readAt rpcs return.
+	backupPhaseDone   = func() {} // call when the entire backup read phase is done.
```

Comment:
These are only used in tests, but it still feels better to put them in the Client rather than globals. Also maybe put "Hook" in the name so the semantics are more clear (that they're just hooks for injecting test synchronization).